flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableOutputFormat.java

```diff
@@ -124,7 +124,7 @@
 	private transient int numNonPartitionColumns;
 
 	// SerDe in Hive-1.2.1 and Hive-2.3.4 can be of different classes, make sure to use a common base class
-	private transient Serializer serializer;
+	private transient Serializer recordSerDe;
```

Comment:
Maybe the type here should be just "Object".

Reply:
It has to be a serializer because we need it to serialize records. Besides, using Object means we have to use reflection to call the `serialize` method. And if we do this for each record, it might hurt performance.